dotCMS/src/integration-test/java/com/dotcms/contenttype/test/ContentResourceTest.java

```diff
@@ -921,6 +1050,34 @@ private void validateChildrenXML(final Map<String, Contentlet> contentletMap,
         }
     }
 
+    private void validateCategoryXML(
+            final String resultXml, final String categoryElement, final Category... categories)
+            throws XPathExpressionException {
+
+        final InputSource xmlSource = new InputSource(new StringReader(
+                resultXml.replaceAll("\\n", "")));
+
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        NodeList itemList = (NodeList) xPath.evaluate("//customTopic/item",
+                xmlSource, XPathConstants.NODESET);
+
+        assertEquals(categories.length, itemList.getLength());
+        for (int i = 0; i < itemList.getLength(); i++) {
+            final Element resultCategory = (Element) itemList.item(i);
+            final String resultCategoryKey = (String) xPath.evaluate("key",
+                    resultCategory, XPathConstants.STRING);
+            Optional<Category> matchCategory = Arrays.stream(categories)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'matchCategory' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885353)